modules/externalipam/api_resource.rb

```diff
@@ -0,0 +1,54 @@
+require 'yaml'
+require 'json'
+require 'net/http'
+require 'uri'
+require 'externalipam/ipam_helper'
+
+module Proxy::Ipam
+  # Class to handle authentication and HTTP transactions with External IPAM providers
+  class ApiResource
+    include ::Proxy::Log
+    include Proxy::Ipam::IpamHelper
+
+    def initialize(params = {})
+      @api_base = params[:api_base]
+      @token = params[:token]
+      @auth_header = params[:auth_header] || 'Authorization'
+    end
+
+    def get(path)
+      uri = URI(@api_base + path)
```

Comment:
Given the repetition in how often you call `URI.encode_www_form()` with this method, should it be solved here?
```suggestion
    def get(path, params = nil)
      url = @api_base + path
      url += "?#{URI.encode_www_form(params)}" if params
      uri = URI(url)
```